app/mutations/application/analystCreateApplicationRevisionStatusMutation.ts

```diff
@@ -0,0 +1,46 @@
+import {graphql} from 'react-relay';
+import RelayModernEnvironment from 'relay-runtime/lib/store/RelayModernEnvironment';
+import {
+  analystCreateApplicationRevisionStatusMutationVariables,
+  analystCreateApplicationRevisionStatusMutation as analystCreateApplicationRevisionStatusMutationType
+} from 'analystCreateApplicationRevisionStatusMutation.graphql';
+import BaseMutation from 'mutations/BaseMutation';
+
+const mutation = graphql`
+  mutation analystCreateApplicationRevisionStatusMutation(
+    $input: CreateApplicationRevisionStatusInput!
+  ) {
+    createApplicationRevisionStatus(input: $input) {
+      applicationRevisionStatus {
+        id
+        versionNumber
+        applicationRevisionByApplicationIdAndVersionNumber {
+          id
+          applicationRevisionStatus {
+            id
```

Comment:
No, that was a few refactors ago. ​Nothing to see here :)

Reply:
I just got confused when re-reading this. I think that our choice of computed column name does not help here.
The first `applicationRevisionStatus` is the object returned by the mutation, but the second `applicationRevisionStatus` is the property from the `ApplicationRevision` type, which was created with a computed column. Naming it `currentApplicationRevisionStatus` would disambiguate that (maybe a good candidate for a tech debt item, and naming convention for future computed columns)

Reply:
Ooh right, because otherwise it would be accessed by something like `applicationRevisionStatusBy...`
Agreed on it being tech debt, it definitely doesn't have to be fixed here

Reply:
I was reading more about graphQL this afternoon, since this query was confusing me 😁

Reply:
> but the second applicationRevisionStatus is the property from the ApplicationRevision type, which was created with a computed column

Agreed, it took me a while even to figure out there was a computed column - and even then, its own status property is named the same thing as the object itself.

It would be a lot simpler just to have a `currentApplicationRevisionStatus` computed column that returns a simple string (of the correct type)?

Reply:
@naomiaro yeah, for me my previous issue was partly due to disbelief at how circular you can get with your queries... in this case nesting a query for an application (and review steps!) inside of application revision status.

And definitely in this case the query appears self-referential, nesting an additional query for `applicationRevisionStatus` inside of a query for `applicationRevisionStatus`. But that's an illusion thanks to the existence of the above-mentioned identically-named computed column on the `applicationRevision`)